.github/workflows/checkstyle-annotate.yml

```
@@ -0,0 +1,18 @@
+name: Run checkstyle
+
+on: pull_request
+
+jobs:
+  checkstyle:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: '11'
+    - uses: kiancross/checkstyle-annotations-action@v1
+    - name: run checkstyle
+      uses: gradle/gradle-build-action@v2.1.2
```

```suggestion
      uses: gradle/gradle-build-action@v2
```

It is better to accept improvements of minor version up.